src/pages/secretary-pages/StudyProgramsPage.tsx

```diff
@@ -5,45 +5,37 @@ import DataPageTemplate from "../templates/DataPageTemplate";
 import { IDataTableHeader } from "../../components/DataTable";
 import { ButtonGroup } from "@chakra-ui/react";
 import AddStudyProgramModal from "../../components/AddStudyProgramModal";
-import { useEffect, useState } from "react";
-import { genericApiGet, studyProgramsEndpoint } from "../../api/GenericApi";
+import { useCallback, useEffect, useState } from "react";
+import { genericApiGet, IStudyProgramsAPI, studyProgramsEndpoint } from "../../api/GenericApi";
+
+const title = "Programe de studiu";
+const headers = [
+  {
+    name: "Nume program de studiu",
+  },
+  {
+    name: "Număr ani",
+  },
+];
+const apiEndpoint = studyProgramsEndpoint;
 
 const Component = () => {
-  ///// Config /////
-  const title = "Programe de studiu";
-  const headers: IDataTableHeader[] = [
-    {
-      name: "Nume program de studiu",
-    },
-    {
-      name: "Număr ani",
-    },
-  ];
-  const apiEndpoint = studyProgramsEndpoint;
+  const [rows, setRows] = useState<IStudyProgramsAPI[]>([]);
```

Comment:
constants should be kept in the component as it will require more focus when reading the code (you have to always asl yourself if this piece of code changes based on some other state variables or not)